lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -377,6 +383,38 @@ def refine_tracks_centroid(tracks, track_width=None):
     track_width_pixels = np.ceil(track_width / tracks._kymo.pixelsize[0])
 
     interpolated_tracks = [track.interpolate() for track in tracks]
+    image = tracks._kymo.get_image(tracks._channel)
+
+    if pixel_threshold is not None:
+        if pixel_threshold <= 0:
+            raise ValueError(f"pixel_threshold should be larger than zero")
+
+        blurred = gaussian_filter(image, [0.5, 0])
```

Comment:
I would really prefer it if we don't duplicate this filter. I fear that if we add this here, we will forget about the fact that there is another one like these in the actual peak detection and they risk going out of sync. I would pull this out into a small subfunction or leave a note on both sides. I'd probably prefer the subfunction way.

Reply:
I was worried about that too, I was just wondering if it was too much for basically a single line function. But I think it's worth it to avoid bugs here. I also thought we should be comparing against the blurred and dilated image, but I'm not 100% sure on this one. I pulled this stuff out into a separate function, what do you think?